language-design.md

```diff
@@ -38,16 +128,50 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+##### A well-designed language:
+
+* has a system or pattern for growth so that it can evolve with the needs of its users.
+* stays true to a specific user, purpose, or philosophy so that development is directed and the language doesn't become obscure because it's trying to be too many things at once.
+* requires just enough syntax to express an idea so that the syntax prescribed is concise and meaningful.
+* "should be easy to use and hard to misuse:" names and functions in the interface of the language are self-documenting, and there are little to no user-accessible functions that have destructive behavior [Bloch, 2006].
+* allows users to expand the language in a way that feels like the underlying language (e.g., supports operator overloading).
+
+##### Programs in a well-designed language:
 
+* "[read] like prose" and are easy to understand--preferably, even by those who are not familiar with the language [Bloch, 2006].
+* has components that can be altered in a predictable way to change the meaning of the program.
+* behave in a predictable way and are obviously correct.
+
+##### A poorly-designed language:
+
+* employs trivial and unnecessary syntax.
+* places a high conceptual burden on the user.
+* requires reading lots of documentation to become familiar with the language.
+
+##### Programs in a poorly-designed language:
+* require excessive error handling.
```

Comment:
I would actually tend to agree that excessive error handling, and I'd add special casing, is a sign of poor design. There are situations like IO and interfacing with networks and other systems that are harder to achieve this goal with, but a language that managed it would be better in that aspect than one that didn't.

Futures and Python's with statement are two examples of ideas that have helped alleviate this problem for networks and IO *slightly*.